bam_checksum.c

```diff
@@ -0,0 +1,1311 @@
+                    goto err;
+                }
+            }
+            continue;
+        }
+
+        if (strncmp(line.s, "# Group", 7) == 0) {
+            // Parse column header so we know which fields are present
+            int n, i = 0, idx;
+            char *ptr = line.s+2;
+            char token[20];
+            while ((n = sscanf(ptr, "%19s%n", token, &idx)) == 1) {
+                if (strcmp(token, "Group") == 0)
+                    header[i] = H_GROUP;
+                else if (strcmp(token, "QC") == 0)
+                    header[i] = H_QC;
+                else if (strcmp(token, "count") == 0)
+                    header[i] = H_COUNT;
+                else if (strcmp(token, "flag+seq") == 0)
+                    header[i] = H_SEQ;
+                else if (strcmp(token, "+name") == 0)
+                    header[i] = H_NAME;
+                else if (strcmp(token, "+qual") == 0)
+                    header[i] = H_QUAL;
+                else if (strcmp(token, "+aux") == 0)
+                    header[i] = H_AUX;
+                else if (strcmp(token, "+chr/pos") == 0)
+                    header[i] = H_POS, o->check_pos = 1;
+                else if (strcmp(token, "+cigar") == 0)
+                    header[i] = H_CIGAR, o->check_cigar = 1;
+                else if (strcmp(token, "+mate") == 0)
+                    header[i] = H_MATE, o->check_mate = 1;
+                else if (strcmp(token, "combined") == 0)
+                    header[i] = H_COMBINED;
+                else  {
+                    fprintf(stderr, "Unrecognised header token '%s'\n", token);
+                    goto err;
+                }
+
+                i++;
+                ptr += idx;
+            }
+            nheader = i;
+
+            continue;
+        }
+
+        if (strncmp(line.s, "# Aux", 5) == 0) {
+            int idx;
+            char c;
+            if (sscanf(line.s, "# Aux tags: %c%n", &c, &idx) == 1)
+                if (!o->tag_str)
+                    o->tag_str = strdup(line.s + idx-1);
+
+            continue;
+        }
+
+        if (strncmp(line.s, "# BAM", 5) == 0) {
+            int idx;
+            char c;
+            if (sscanf(line.s, "# BAM flags: %c%n", &c, &idx) == 1)
+                o->flag_mask = bam_str2flag(line.s + idx-1);
+
+            continue;
+        }
+
+        if (!line.l || *line.s == '#')
+            continue;
+
+
+        // Header done.  Now parse the data lines
+        if (strncmp(line.s, "all ", 4) == 0 ||
+            strncmp(line.s, "all\t", 4) == 0)
+            continue;
+
+        char col[11][128], *ptr = line.s;
+        int nf;
+        for (nf = 0; nf < 11; nf++) {
+            int idx;
+            int n = sscanf(ptr, "%127s%n", col[nf], &idx);
+            if (n <= 0)
+                break;
+            if (strlen(col[nf]) == 127) {
+                fprintf(stderr, "Field too long\n");
+                goto err;
+            }
+            ptr += n == 1 ? idx : 0;
```

Comment:
Can `n` ever not be 1 here?

Reply:
Indeed I think not.  It probably dates to before the `n <= 0` bit was added above.